proofs/eo/cpc/programs/PolyNorm.eo

```diff
@@ -0,0 +1,204 @@
+    (($mvar_mul_mvar (* a1 a2) 1)          (* a1 a2))
+    (($mvar_mul_mvar 1 (* b1 b2))          (* b1 b2))
+    (($mvar_mul_mvar 1 1)                  1)
+  )
+)
+
+; program: $mon_mul_mon
+; args:
+; - a @Monomial: The first monomial to multiply.
+; - b @Monomial: The second monomial to multiply.
+; return: the multiplication of the given monomials.
+(program $mon_mul_mon ((T Type) (U Type) (a1 T) (a2 U) (c1 Real) (c2 Real))
+  (@Monomial @Monomial) @Monomial
+  (
+    (($mon_mul_mon (@mon a1 c1) (@mon a2 c2))  (@mon ($mvar_mul_mvar a1 a2) (eo::mul c1 c2)))
+  )
+)
+
+; program: $poly_mul_mon
+; args:
+; - a @Monomial: The monomial to multiply.
+; - p @Polynomial: The polynomial to multiply.
+; return: the multiplication of the polynomial by a monomial.
+(program $poly_mul_mon ((m1 @Monomial) (m2 @Monomial) (p2 @Polynomial :list))
+  (@Monomial @Polynomial) @Polynomial
+  (
+    (($poly_mul_mon m1 (@poly m2 p2)) ($poly_add (@poly ($mon_mul_mon m1 m2)) ($poly_mul_mon m1 p2)))   ; NOTE: this amounts to an insertion sort
+    (($poly_mul_mon m1 @poly.zero)    @poly.zero)
+  )
+)
+
+; program: $poly_mul
+; args:
+; - p1 @Polynomial: The first polynomial to multiply.
+; - p2 @Polynomial: The second polynomial to multiply.
+; return: the multiplication of the given polynomials.
+(program $poly_mul ((m @Monomial) (p1 @Polynomial :list) (p @Polynomial))
+  (@Polynomial @Polynomial) @Polynomial
+  (
+    (($poly_mul (@poly m p1) p) ($poly_add ($poly_mul_mon m p) ($poly_mul p1 p)))
+    (($poly_mul @poly.zero p)   @poly.zero)
+    (($poly_mul p @poly.zero)   @poly.zero)
+  )
+)
+
+; program: $get_poly_norm
+; args:
+; - a T: The arithmetic term to process of type Int or Real.
+; return: the polynomial corresponding to the (normalized) form of a.
+(program $get_poly_norm ((T Type) (U Type) (V Type) (a T) (a1 U) (a2 V :list))
+  (T) @Polynomial
+  (
+    (($get_poly_norm (- a1))       ($poly_neg ($get_poly_norm a1)))
+    (($get_poly_norm (+ a1 a2))    ($poly_add ($get_poly_norm a1) ($get_poly_norm a2)))
+    (($get_poly_norm (- a1 a2))    ($poly_add ($get_poly_norm a1) ($poly_neg ($get_poly_norm a2))))
+    (($get_poly_norm (* a1 a2))    ($poly_mul ($get_poly_norm a1) ($get_poly_norm a2)))
+    (($get_poly_norm (to_real a1)) ($get_poly_norm a1))
+    (($get_poly_norm a)            (eo::define ((aq (eo::to_q a)))
+                                  ; if it is a constant, which can be tested if to_q is idempotent after the first
+                                  (eo::ite (eo::is_eq aq (eo::to_q aq))
+                                    ; if it is zero, it cancels, otherwise it is 1 with itself as coefficient
+                                    (eo::ite (eo::is_eq aq 0/1)
+                                      @poly.zero
+                                      (@poly (@mon 1 aq)))
+                                  (@poly (@mon (* a) 1/1)))))    ; introduces list
+  )
+)
+
+; define: $is_arith_poly_norm
```

Comment:
The function's name is `$is_poly_norm`. So, either this line or line 132 should change.